src/api/core/organizations.rs

```diff
@@ -1546,6 +1562,19 @@ async fn put_policy(
         }
     }
 
+    //This check is required since invited users automatically get accepted if mailing is not enabled (this seems like a vaultwarden specific feature)
+    //As a result of this the necessary "/accepted"-endpoint doesn't get hit.
+    //But this endpoint is required for autoenrollment while invitation.
+    //Nevertheless reset password is fully fuctiontional in settings without mailing by manual enrollment
+    if pol_type_enum == OrgPolicyType::ResetPassword && data.enabled {
+        //Maybe this check could get improved to get away from string checking
+        //Deserializing from "serde_json::from_value" won't work with current scope
+        let pol_add = serde_json::to_string(&data.data)?;
+        if pol_add.contains("\"autoEnrollEnabled\":true") && !CONFIG.mail_enabled() {
+            err!("Autoenroll can't be used since it requires enabled emailing")
+        }
+    }
+
```

Comment:
I already added a check on email-disabled instances. While setting the policy, an error is thrown if auto-enrollment is enabled on those instances preventing its usage

Reply:
This is probably a nicer option. Which also does some type-checking.
It also prevents from checking the `policy_data` if mail is enabled. This saves some checking and converting cycles.

```suggestion
    if pol_type_enum == OrgPolicyType::ResetPassword && data.enabled && !CONFIG.mail_enabled() {
        if let Some(policy_data) = &data.data {
            if policy_data["autoEnrollEnabled"].as_bool().unwrap_or(false) {
                err!("Autoenroll can't be used since it requires enabled emailing")
            }
        }
    }
```

Reply:
I had something similar in the first place but I struggeld a bit with Rust.
Thanks for the improvement. I took it as you suggested